lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -0,0 +1,174 @@
+"""
+pbench-clear-tools
+
+This script will remove tools that have been registered.  If no options are
+used, then all tools from the "default" tool group are removed.  Specifying
+a tool name and/or remote host will limit the scope of the removal.
+
+
+"""
+import shutil
+import sys
+import os
+import pathlib
+
+import click
+
+from pbench.agent.utils import setup_logging
+from pbench.cli.agent import CliContext, pass_cli_context
+from pbench.cli.agent.commands.tools.base import ToolCommand
+from pbench.cli.agent.options import common_options
+
+
+class ClearTools(ToolCommand):
+    """Clear registered tools"""
+
+    def __init__(self, context):
+        super().__init__(context)
+
+        self.logger = setup_logging(logfile=self.pbench_log)
+
+    def execute(self):
+        errors = 0
+
+        self.tool_group_dir = self.tool_group_dir(self.context.group)
+        if not self.tool_group_dir.exists():
+            return 1
+
+        try:
+            self.context.remote = self.context.remote.split(",")
```

Comment:
Seems like we should not be modifying the existing `click` context, but using a local variable here?  Or do we need the modified `click` context for some reason?

Reply:
fixed